src/contentChunk.js

```diff
@@ -95,12 +95,34 @@ export const promisesToRemoveAllTransactionBodies = (tx) => {
       removeBodyPromises.push(() => removeBodyById(tx.response.bodyId))
     }
 
-    if (tx.orchestrations && tx.orchestrations.length > 0) {
-      for (let orch of tx.orchestrations) {
-        try {
-          removeBodyPromises = removeBodyPromises.concat(await promisesToRemoveAllTransactionBodies(orch))
-        } catch (err) {
-          return reject(err)
+    if (tx.orchestrations) {
+      if (typeof tx.orchestrations === 'object') {
```

Comment:
Im not sure if this check is really needed here. Based on the transaction schema model, the orchestrations/routes will always be an array.
https://github.com/jembi/openhim-core-js/blob/master/src/model/transactions.js#L69-L70

The removal of transaction bodies are based on the record that is saved within the database which will always be an array for orchestrations/routes.

Reply:
I think where some of the confusion might come in as with the `transactionUpdate` method which can do `$push` updates of the document in either an array, or an object format

Reply:
okay, will fix this now.